Kitodo/src/main/java/org/kitodo/production/helper/IndexWorker.java

```diff
@@ -50,91 +42,64 @@ public IndexWorker(SearchService searchService) {
      * @param searchService
      *            SearchService instance used for indexing
      */
-    public IndexWorker(SearchService searchService, int startIndexing) {
+    public IndexWorker(SearchService searchService, ObjectType objectType, IndexWorkerStatus indexWorkerStatus, boolean indexAllObjects) {
         this.searchService = searchService;
-        this.startIndexing = startIndexing;
+        this.indexWorkerStatus = indexWorkerStatus;
+        this.indexAllObjects = indexAllObjects;
+        this.objectType = objectType;
     }
 
     @Override
     @SuppressWarnings("unchecked")
     public void run() {
-        this.indexedObjects = 0;
+        int maxAttempts = ConfigCore.getIntParameterOrDefaultValue(ParameterCore.ELASTICSEARCH_ATTEMPTS);
         int batchSize = ConfigCore.getIntParameterOrDefaultValue(ParameterCore.ELASTICSEARCH_BATCH);
-        int indexLimit = ConfigCore.getIntParameterOrDefaultValue(ParameterCore.ELASTICSEARCH_INDEXLIMIT);
-        try {
-            int amountToIndex = getAmountToIndex();
-
-            if (amountToIndex < batchSize) {
-                if (indexAllObjects) {
-                    indexObjects(searchService.getAll(this.startIndexing, amountToIndex));
-                } else {
-                    indexObjects(searchService.getAllNotIndexed(this.startIndexing, amountToIndex));
-                }
-            } else {
-                if (amountToIndex > indexLimit) {
-                    amountToIndex = indexLimit;
-                }
-                while (this.indexedObjects < amountToIndex) {
-                    indexChunks(batchSize);
+        int timeBetweenAttempts = ConfigCore.getIntParameterOrDefaultValue(ParameterCore.ELASTICSEARCH_TIME_BETWEEN_ATTEMPTS);
+        int maxBatch = indexWorkerStatus.getMaxBatch();
+
+        Integer nextBatch = indexWorkerStatus.getAndIncrementNextBatch();
+        while (nextBatch < maxBatch) {
+            // nextBatch is a valid batch that needs to be processed
+
+            int attempt = 1;
+            while (attempt < maxAttempts) {
+                try {
+                    int offset = nextBatch * batchSize;
+                    logger.info("index " + objectType.toString() + " with offset " + offset + " and attempt " 
+                        + attempt + "/" + maxAttempts);
+
+                    if (indexAllObjects) {
+                        indexObjects(searchService.getAll(offset, batchSize));
+                    } else {
+                        indexObjects(searchService.getAllNotIndexed(offset, batchSize));
+                    }
+
+                    break;
+                } catch (Exception e) {
+                    logger.error(e.getMessage(), e);
+                    attempt += 1;
+                    try {
+                        Thread.sleep(timeBetweenAttempts);
+                    } catch (InterruptedException e2) {
+                        // ignore
```

Comment:
Add maybe a log trace entry instead of silence ignoring this? I know this should not happen in a regular way but if this is happen then you have at least the change to get notified.

Reply:
Done